index.bs

```diff
@@ -2364,41 +2409,30 @@ It takes the following input parameters:
-    match occurs if a credential matches <code>|rpId|</code> and an |allowCredentialDescriptorList| [=list/item=]'s
-    {{PublicKeyCredentialDescriptor/id}} and {{PublicKeyCredentialDescriptor/type}} members.
-1. If |allowCredentialDescriptorList| is now [=list/empty=], return an error code equivalent to "{{NotAllowedError}}" and
-    terminate the operation.
-
-1. Let |selectedCredential| be a [=public key credential|credential=] as follows. If the [=list/size=] of
-    |allowCredentialDescriptorList|
-
-    <dl class="switch">
-        :   is exactly 1
-        ::  Let |selectedCredential| be the [=public key credential|credential=] matching
-            <code>|allowCredentialDescriptorList|[0]</code>.
-
-        :   is greater than 1
-        ::  Prompt the user to select |selectedCredential| from the [=public key credential|credentials=] matching the
-            [=list/items=] in |allowCredentialDescriptorList|.
-    </dl>
-
-
-1. Obtain [=user consent=] for using |selectedCredential|. The prompt for obtaining this [=user consent|consent=] MAY be shown by
-    the [=authenticator=] if it has its own output capability, or by the user agent otherwise. The prompt SHOULD display the
-    |rpId| and any additional displayable data associated with |selectedCredential|, if possible.
+1. Let |credentialOptions| be a new empty [=set=] of [=public key credential sources=].
+1. If |allowCredentialDescriptorList| was supplied, then [=list/for each=] |descriptor| of |allowCredentialDescriptorList|:
+    1. Let |credSource| be the result of [=looking up=] <code>|descriptor|.{{PublicKeyCredentialDescriptor/id}}</code> in this
+        authenticator.
+    1. If |credSource| is not `null`, [=set/append=] it to |credentialOptions|.
+1. Otherwise (|allowCredentialDescriptorList| was not supplied), [=map/for each=] <var ignore>key</var> → |credSource| of this
+    authenticator's [=credentials map=], [=set/append=] |credSource| to |credentialOptions|.
+1. [=list/Remove=] any items from |credentialOptions| whose [=public key credential source/rpId=] is not equal to
+    |rpId|.
+1. If |credentialOptions| is now empty, return an error code equivalent to "{{NotAllowedError}}" and terminate the operation.
+
+1. Prompt the user to select a [=public key credential source=] |selectedCredential| from |credentialOptions|.
+    Obtain [=user consent=] for using |selectedCredential|. The prompt for obtaining this [=user consent|consent=] may be shown
+    by the [=authenticator=] if it has its own output capability, or by the user agent otherwise.
 
     If |requireUserVerification| is `true`, the method of obtaining [=user consent=] MUST include [=user verification=].
 
-    If |requireUserPresence| is `true`, the method of obtaining [=user consent=] MUST include a [=test of user presence=].
+    If <var ignore>requireUserPresence</var> is `true`, the method of obtaining [=user consent=] MUST include a 
```

Comment:
This `<var>` is now defined and `ignore`d in two places... :)

Reply:
fixed